src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1154,14 +1153,23 @@ internal SingleLookupResult CheckViability(Symbol symbol, int arity, LookupOptio
                                         ref useSiteDiagnostics,
                                         basesBeingResolved))
             {
-                if (inaccessibleViaQualifier)
-                {
-                    diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadProtectedAccess, unwrappedSymbol, accessThroughType, this.ContainingType) : null;
-                }
-                else
-                {
-                    var unwrappedSymbols = ImmutableArray.Create<Symbol>(unwrappedSymbol);
-                    diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadAccess, new[] { unwrappedSymbol }, unwrappedSymbols, additionalLocations: ImmutableArray<Location>.Empty) : null;
+                 if (!diagnose)
+                 {
+                    diagInfo = null;
+                 }
+                 else
+                 {
+                    if (inaccessibleViaQualifier)
+                    {
+                       diagInfo = new CSDiagnosticInfo(ErrorCode.ERR_BadProtectedAccess, unwrappedSymbol, accessThroughType, this.ContainingType);
+                    }
+                    else if (getFriendRefNotEqualToThis())
```

Comment:
We usually prefix `bool`-returning functions and properties with words like `Is` or `Has`. I would call this function something like `isBadIvtSpecification`.